pkg/testcase/upgradesuc.go

```diff
@@ -51,12 +49,40 @@ func TestUpgradeClusterSUC(cluster *shared.Cluster, k8sClient *k8s.Client, versi
 		return fmt.Errorf("failed to write file: %s", err)
 	}
 
-	workloadErr = shared.ManageWorkload("apply", "plan.yaml")
-	Expect(workloadErr).NotTo(HaveOccurred(), "failed to upgrade cluster.")
+	planApplyErr := shared.ManageWorkload("apply", "plan.yaml")
+	Expect(planApplyErr).NotTo(HaveOccurred(), "failed to upgrade cluster - apply plan.yaml step failed.")
 
 	ok, err := k8sClient.CheckClusterHealth(0)
 	Expect(err).NotTo(HaveOccurred(), err, "error checking cluster health")
 	Expect(ok).To(BeTrue(), "cluster health check failed")
 
 	return nil
 }
+
+func applySucYamls() {
+	sucUrl := "https://github.com/rancher/system-upgrade-controller/releases/latest/download/system-upgrade-controller.yaml"
+	sucCRDUrl := "https://github.com/rancher/system-upgrade-controller/releases/latest/download/crd.yaml"
+
+	shared.LogLevel("info", "Applying system-upgrade-controller manifest from url: %s", sucUrl)
+	applyErr := shared.ApplyWorkloadURL(sucUrl)
+	if applyErr != nil {
+		shared.LogLevel(
+			"warn", "error applying system-upgrade-controller manifest from url: %s error: %v", sucUrl, applyErr)
+		shared.LogLevel("debug", "applying system-upgrade-controller manifest from local file")
+		// Fallback to local file if URL fails
+		applyErr = shared.ManageWorkload("apply", "suc.yaml")
```

Comment:
Once we apply the workload, can we assert that the pods are in Running state? I think we should make that the standard